lib/mysql.js

```diff
@@ -412,7 +412,7 @@ MySQL.prototype.fromColumnValue = function(prop, val) {
         // MySQL allows, unless NO_ZERO_DATE is set, dummy date/time entries
         // new Date() will return Invalid Date for those, so we need to handle
         // those separate.
-        if (!val || val == '0000-00-00 00:00:00' || val == '0000-00-00') {
+        if (!val || /^0{4}(-00){2}( (00:){2}0{2}(\.0{1,6}){0,1}){0,1}$/.test(val)) {
```

Comment:
Ah, I get it! The `test()` method checks if the string specified in the function argument matches the regex. It will return true or false. See also [this article on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test).